### ISSUE TITLE
PyTorch integration

### ISSUE THREAD

**Issue by wcqc** [gstenzel/qandle#1]
Hi, 

Does this support dynamically structured and defined computation graphs? For example, constructing a quantum circuit on the fly in the forward pass of PyTorch? 

Thanks for any clarification.

**Comment by gstenzel** [comment]
This module draws most of its advantage in runtime speed by optimizing during `__init__`. Therefore, creating circuits during the forward pass is not recommended with this package. You might rather want look towards PennyLane for such use-case.

**Comment by wcqc** [comment]
How about reusing a predefined quantum circuit in the forward pass but feeding in different inputs in forward?

**Comment by gstenzel** [comment]
Yes, that is possible using named gates. Check out [the docs](https://gstenzel.github.io/qandle/tutorial/01helloquantum.html#define-a-quantum-circuit): define a gate, e.g. `RX(qubit=1, name="gate1")`, and pass custom values to it during the forward pass of your module `circuit(gate1=torch.tensor(1.2))`. This should support batched inputs out-of-the-box.
